edition = "2023";

package mmm.v1.marketing.analysis;

option features.field_presence = IMPLICIT;
option java_multiple_files = true;

message ResponsePoint {
  // The amount of the input that drives the incremental KPI. Required.
  double input_value = 1;

  // The incremental KPI caused by the input. Required.
  double incremental_kpi = 2;
}

// A response curve is used to show how much incremental KPI moves in response
// to changes to the input value (e.g. spend amount on a paid media channel,
// advertising impressions in a channel, etc).
message ResponseCurve {
  // The name of the input. Required.
  string input_name = 1;

  // The response points. Required.
  repeated ResponsePoint response_points = 2;
}
